src/main/java/com/epam/ta/reportportal/core/item/impl/history/TestItemsHistoryHandlerImpl.java

```diff
@@ -56,18 +62,35 @@
 
 	private LaunchRepository launchRepository;
 
+	private GetTestItemHandler getTestItemHandler;
+
+	private TestItemResourceAssembler itemResourceAssembler;
+
 	@Autowired
-	public TestItemsHistoryHandlerImpl(TestItemRepository testItemRepository, LaunchRepository launchRepository) {
+	public TestItemsHistoryHandlerImpl(TestItemRepository testItemRepository, LaunchRepository launchRepository,
+			GetTestItemHandler getTestItemHandler, TestItemResourceAssembler itemResourceAssembler) {
 		this.testItemRepository = testItemRepository;
 		this.launchRepository = launchRepository;
+		this.getTestItemHandler = getTestItemHandler;
+		this.itemResourceAssembler = itemResourceAssembler;
 	}
 
 	@Override
-	public List<TestItemHistoryElement> getItemsHistory(ReportPortalUser.ProjectDetails projectDetails, Long[] startPointsIds,
-			int historyDepth, boolean showBrokenLaunches) {
+	public List<TestItemHistoryElement> getItemsHistory(ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user,
+			Queryable filter, Pageable pageable, @Nullable Long launchId, @Nullable Long filterId, int launchesLimit, int historyDepth) {
 
 		validateHistoryDepth(historyDepth);
-		List<Long> itemIds = Lists.newArrayList(startPointsIds);
+		Page<Long> testItems = getTestItemHandler.getTestItemsIds(
+				filter,
+				pageable,
+				projectDetails,
+				user,
+				launchId,
+				filterId,
+				launchesLimit
+		);
+		List<Long> itemIds = new ArrayList<>(testItems.getContent());
```

Comment:
getContent returns List, so new List creation isn't necessary

Reply:
It's because getContent returns unmodifiable list. removeAll method does not work on it